iceoryx_posh/include/iceoryx_posh/popo/sample.hpp

```diff
@@ -48,6 +48,21 @@ class Sample
     ///
     T* operator->() noexcept;
 
+    ///
+    /// @brief operator* Provide a reference to the encapsulated type.
+    /// @return A T& to the encapsulated type.
+    /// @details Only available for non void type T.
+    ///
+    template <typename S = T,
```

Comment:
We can probably eliminate this when we no longer instantiate Sample<void>. (Done for the current API). It is the only reason for this SFINAE magic right now.